context/app/static/js/components/bulkDownload/hooks.ts

```diff
@@ -0,0 +1,199 @@
+import { z } from 'zod';
+
+import { bulkDownloadOptionsField, bulkDownloadMetadataField } from 'js/components/bulkDownload/bulkDownloadFormFields';
+import useBulkDownloadToasts from 'js/components/bulkDownload/toastHooks';
+import { ALL_BULK_DOWNLOAD_OPTIONS } from 'js/components/bulkDownload/constants';
+import { BulkDownloadDataset, useBulkDownloadStore } from 'js/stores/useBulkDownloadStore';
+import { useSearchHits } from 'js/hooks/useSearchData';
+import { useProtectedDatasetsForm } from 'js/hooks/useProtectedDatasets';
+import { createDownloadUrl } from 'js/helpers/functions';
+import { checkAndDownloadFile, postAndDownloadFile } from 'js/helpers/download';
+import { trackEvent } from 'js/helpers/trackers';
+import { getIDsQuery } from 'js/helpers/queries';
+
+const schema = z
+  .object({
+    ...bulkDownloadOptionsField,
+    ...bulkDownloadMetadataField,
+  })
+  .partial()
+  .required({ bulkDownloadOptions: true });
+
+export interface BulkDownloadFormTypes {
+  bulkDownloadOptions: string[];
+  bulkDownloadMetadata: boolean;
+}
+
+function useBulkDownloadForm() {
+  const {
+    handleSubmit,
+    control,
+    reset,
+    setValue,
+    formState: { errors },
+    trigger,
+  } = useForm<BulkDownloadFormTypes>({
+    defaultValues: {
+      bulkDownloadOptions: [],
+      bulkDownloadMetadata: false,
+    },
+    mode: 'onChange',
+    resolver: zodResolver(schema),
+  });
+
+  return {
+    handleSubmit,
+    setValue,
+    control,
+    errors,
+    reset,
+    trigger,
+  };
+}
+
+interface BulkDownloadDialogProps {
+  deselectRows?: (uuids: string[]) => void;
+}
+function useBulkDownloadDialog({ deselectRows }: BulkDownloadDialogProps) {
+  const { isOpen, uuids, open, close, setUuids } = useBulkDownloadStore();
+  const { control, handleSubmit, errors, reset, trigger } = useBulkDownloadForm();
+  const { toastErrorDownloadFile, toastSuccessDownloadFile } = useBulkDownloadToasts();
+
+  // Fetch datasets for the selected uuids
+  const datasetQuery = {
+    query: getIDsQuery([...uuids]),
+    _source: ['hubmap_id', 'processing', 'uuid', 'files', 'processing_type'],
+    size: 1000,
+  };
+  const { searchHits, isLoading } = useSearchHits<BulkDownloadDataset>(datasetQuery);
+  const datasets = searchHits.map(({ _source }) => _source);
+
+  // Which options to show in the dialog
+  const downloadOptions = ALL_BULK_DOWNLOAD_OPTIONS.map((option) => ({
+    ...option,
+    count: datasets.filter((dataset) => option.isIncluded(dataset)).length,
+  })).filter((option) => option.count > 0);
+
+  // Remove selected uuids from the list and deselect them in the table if needed
+  const removeUuidsOrRows = useCallback(
+    (uuidsToRemove: string[]) => {
+      if (deselectRows) {
+        deselectRows([...uuidsToRemove]);
+      }
+      setUuids(new Set([...uuids].filter((uuid) => !uuidsToRemove.includes(uuid))));
+    },
+    [deselectRows, setUuids, uuids],
+  );
+
+  const { ...protectedDatasetsFields } = useProtectedDatasetsForm({
```

Comment:
The extra destructuring here doesn't seem necessary, we can reintroduce it in the future if we need to extract a specific field and keep it separate from the `protectedDatasetsFields` object. 

```suggestion
  const protectedDatasetsFields = useProtectedDatasetsForm({
```